packages/ember-htmlbars/tests/integration/component_invocation_test.js

```diff
@@ -451,3 +471,234 @@ QUnit.test('implementing `render` allows pushing into a string buffer', function
+
+  run(function() {
+    view.set('someProp', 'tomdale');
+  });
+
+  equal(jQuery('#qunit-fixture').text(), 'In layout - someProp: tomdale');
+  equal(didReceiveAttrs, 2, "The didReceiveAttrs hook fired again");
+  equal(willUpdate, 1, "The willUpdate hook fired once");
+
+  Ember.run(view, 'rerender');
+
+  equal(jQuery('#qunit-fixture').text(), 'In layout - someProp: tomdale');
+  equal(didReceiveAttrs, 3, "The didReceiveAttrs hook fired again");
+  equal(willUpdate, 2, "The willUpdate hook fired again");
+});
+
+QUnit.test('block with properties on attrs', function() {
+  registry.register('template:components/with-block', compile('In layout - someProp: {{attrs.someProp}} - {{yield}}'));
+
+  view = appendViewFor('<with-block someProp="something here">In template</with-block>');
+
+  equal(jQuery('#qunit-fixture').text(), 'In layout - someProp: something here - In template');
+});
+
+QUnit.test('moduleName is available on _renderNode when a layout is present', function() {
+  expect(1);
+
+  var layoutModuleName = 'my-app-name/templates/components/sample-component';
+  var sampleComponentLayout = compile('Sample Component - {{yield}}', {
+    moduleName: layoutModuleName
+  });
+  registry.register('template:components/sample-component', sampleComponentLayout);
+  registry.register('component:sample-component', Component.extend({
+    didInsertElement: function() {
+      equal(this._renderNode.lastResult.template.meta.moduleName, layoutModuleName);
+    }
+  }));
+
+  view = EmberView.extend({
+    layout: compile('<sample-component />'),
+    container
+  }).create();
+
+  runAppend(view);
+});
+
+QUnit.test('moduleName is available on _renderNode when no layout is present', function() {
+  expect(1);
+
+  var templateModuleName = 'my-app-name/templates/application';
+  registry.register('component:sample-component', Component.extend({
+    didInsertElement: function() {
+      equal(this._renderNode.lastResult.template.meta.moduleName, templateModuleName);
+    }
+  }));
+
+  view = EmberView.extend({
+    layout: compile('{{#sample-component}}Derp{{/sample-component}}', {
+      moduleName: templateModuleName
+    }),
+    container
+  }).create();
+
+  runAppend(view);
+});
+
+QUnit.test('parameterized hasBlock default', function() {
+  registry.register('template:components/check-block', compile('{{#if (hasBlock)}}Yes{{else}}No{{/if}}'));
+
+  view = appendViewFor('<check-block id="expect-yes-1" />  <check-block id="expect-yes-2"></check-block>');
+
+  equal(view.$('#expect-yes-1').text(), 'Yes');
+  equal(view.$('#expect-yes-2').text(), 'Yes');
+});
+
+QUnit.test('non-expression hasBlock ', function() {
+  registry.register('template:components/check-block', compile('{{#if hasBlock}}Yes{{else}}No{{/if}}'));
+
+  view = appendViewFor('<check-block id="expect-yes-1" />  <check-block id="expect-yes-2"></check-block>');
+
+  equal(view.$('#expect-yes-1').text(), 'Yes');
```

Comment:
Just discussed with @wycats and (and others), I believe that my prior statement is incorrect and that we will differentiate between (`hasBlock` should be false):

 `<check-block id="expect-yes-1" />`

And (`hasBlock` should be true):

`<check-block id="expect-yes-1"></check-block>`

Sorry for the confusion...